codelists/views/version_diff.py

```diff
@@ -48,27 +50,36 @@ def version_diff(request, clv, other_tag_or_hash):
         "lhs_only_codes": lhs_only_codes,
         "rhs_only_codes": rhs_only_codes,
         "common_codes": common_codes,
-        "lhs_only_summary": summarise(lhs_only_codes, lhs_coding_system),
-        "rhs_only_summary": summarise(rhs_only_codes, rhs_coding_system),
-        "common_summary": summarise(common_codes, lhs_coding_system),
+        "lhs_only_summary": summarise(
+            lhs_only_codes, lhs_coding_system, lhs_csv_data_codes_to_terms
+        ),
+        "rhs_only_summary": summarise(
+            rhs_only_codes, rhs_coding_system, rhs_csv_data_codes_to_terms
+        ),
+        "common_summary": summarise(
+            common_codes, lhs_coding_system, lhs_csv_data_codes_to_terms
+        ),
     }
 
     return render(request, "codelists/version_diff.html", ctx)
 
 
-def summarise(codes, coding_system):
+def summarise(codes, coding_system, csv_data_codes_to_terms=None):
+    csv_data_codes_to_terms = csv_data_codes_to_terms or {}
     code_to_term = coding_system.code_to_term(codes)
 
+    def get_term(code):
+        term = code_to_term[code]
+        if term == "Unknown" and csv_data_codes_to_terms.get(code) is not None:
+            term = f"[Unknown] {csv_data_codes_to_terms[code]}"
+        return term
+
     if coding_system.id == "dmd":
         # dm+d has no hierarchy to look up, just return the codes themselves with their
         # terms
-        summary = [
-            {
-                "code": code,
-                "term": code_to_term[code],
-            }
-            for code in codes
-        ]
+        summary = []
```

Comment:
any particular reason for changing away from a list comp here?

Reply:
No - originally I wrote the new code for getting the term without the inner `get_term` function, and it was more readable without the list comprehension, and then I didn't put it back again.  I have now :)